src/main/java/io/openliberty/tools/intellij/lsp4jakarta/lsp4ij/persistence/PersistenceEntityQuickFix.java

```diff
@@ -54,59 +63,65 @@
  * @author Leslie Dawson (lamminade)
  *
  */
-public class PersistenceEntityQuickFix {
+public class PersistenceEntityQuickFix implements IJavaCodeActionParticipant {
+    private static final Logger LOGGER = Logger.getLogger(PersistenceEntityQuickFix.class.getName());
+
+    @Override
+    public String getParticipantId() {
+        return PersistenceEntityQuickFix.class.getName();
+    }
+
     public List<? extends CodeAction> getCodeActions(JavaCodeActionContext context, Diagnostic diagnostic) {
-        List<CodeAction> codeActions = new ArrayList<>();
-        PsiElement node = context.getCoveredNode();
-        PsiClass parentType = getBinding(node);
+        return addConstructor(diagnostic, context);
+    }
+
+    @Override
+    public CodeAction resolveCodeAction(JavaCodeActionResolveContext context) {
+        final CodeAction toResolve = context.getUnresolved();
+        final PsiElement node = context.getCoveredNode();
+        final PsiClass parentType = PsiTreeUtil.getParentOfType(node, PsiClass.class);
+
         if (parentType != null) {
```

Comment:
It seems this check was being made before in getCodeActions() to determine whether a code action should even be created and has been moved into resolveCodeAction(). This seems to be too late. I think this might produce a quick fix that does nothing when the user clicks on it (if parentType == null), whereas before no quick fix would be presented to the user.